Modules/mathmodule.c

```diff
@@ -2490,6 +2490,151 @@ math_isclose_impl(PyObject *module, double a, double b, double rel_tol,
 }
 
 
+/*[clinic input]
+math.comb
+
+    n: object
+
+    k: object
+    /
+
+Return the binomial coefficient indexed by the pair of integers n >= k >= 0.
+
+It is the coefficient of kth term in polynomial expansion of the expression
+(1 + x)^n. It is also known as the number of ways to choose an unordered
+subset of k elements from a fixed set of n elements, usually called
+*n choose k*.
+
+Raises a TypeError if argument(s) are non-integer and ValueError
+if argument(s) are negative or k > n.
+
+[clinic start generated code]*/
+
+static PyObject *
+math_comb_impl(PyObject *module, PyObject *n, PyObject *k)
+/*[clinic end generated code: output=bd2cec8d854f3493 input=75e1a19623bae7dc]*/
+{
+    if (!(PyLong_Check(n) && PyLong_Check(k))) {
+        PyErr_SetString(PyExc_TypeError,
+            "comb() only accepts integer arguments");
+        return NULL;
+    }
+    n = PyNumber_Long(n);
+    if (n == NULL) {
+        return NULL;
+    }
+    k = PyNumber_Long(k);
+    if (k == NULL) {
+        Py_DECREF(n);
+        return NULL;
+    }
+
+    PyObject *val = NULL,
+        *temp_obj1 = NULL,
+        *temp_obj2 = NULL,
+        *dump_var = NULL;
+    int overflow, cmp;
+    long long i, terms;
+
+    cmp = PyObject_RichCompareBool(n, k, Py_LT);
+    if (cmp == 1) {
+        PyErr_Format(PyExc_ValueError,
+                     "n must be an integer >= k");
+        goto fail_comb;
+    }
+    else if (cmp == -1) {
```

Comment:
Minor nit:  We usually put the -1 one test first.  Also, there is a mild benefit to testing "cmp > 0" vs "cmp == 1" and "cmp < 0" vs "cmp == -1".

Reply:
Made the changes thanks.